src/common/libjob/job.c

```diff
@@ -84,9 +100,15 @@ flux_future_t *flux_job_submit (flux_t *h, const char *jobspec, int priority,
     if (!(flags & FLUX_JOB_PRE_SIGNED)) {
 #if HAVE_FLUX_SECURITY
         flux_security_t *sec;
+        const char *mech = NULL;
+        uint32_t owner;
+
+        if (attr_get_u32 (h, "security.owner", &owner) == 0
+                                                    && getuid () == owner)
+            mech = "none";
```

Comment:
There's some whitespace oddness here:

```suggestion
        if (attr_get_u32 (h, "security.owner", &owner) == 0
            && getuid () == owner)
            mech = "none";
```

Also, though it is pretty clear now, I always feel like it is a good idea to over-comment security significant code. In this case we might want to add a comment that states the reason behind switching to the 'none' signing mechanism, to avoid even slightly perplexing future code sleuths. Just a suggestion though!